Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/CalculateMonteCarloAbsorption.py

```diff
@@ -0,0 +1,796 @@
+                if self._container_inner_radius <= self._sample_radius:
+                    issues['ContainerInnerRadius'] = 'Must be greater than SampleRadius'
+                if self._container_outer_radius <= self._container_inner_radius:
+                    issues['ContainerOuterRadius'] = 'Must be greater than ContainerInnerRadius'
+
+            if self._shape == 'Annulus':
+                if self._container_inner_radius >= self._sample_inner_radius:
+                    issues['ContainerInnerRadius'] = 'Must be less than SampleInnerRadius'
+                if self._container_outer_radius <= self._sample_outer_radius:
+                    issues['ContainerOuterRadius'] = 'Must be greater than SampleOuterRadius'
+
+        return issues
+
+    def _get_efixed(self):
+        """
+        Returns the efixed value relating to the specified workspace
+        """
+        inst = self._sample_ws.getInstrument()
+
+        if inst.hasParameter('Efixed'):
+            return inst.getNumberParameter('Efixed')[0]
+
+        if inst.hasParameter('analyser'):
+            analyser_comp = inst.getComponentByName(inst.getStringParameter('analyser')[0])
+
+            if analyser_comp is not None and analyser_comp.hasParameter('Efixed'):
+                return analyser_comp.getNumberParameter('EFixed')[0]
+
+        raise ValueError('No Efixed parameter found')
+
+    # ------------------------------- Converting to/from wavelength -------------------------------
+
+    def _convert_to_wavelength(self, workspace):
+        """
+        Converts the specified workspace to units of wavelength.
+
+        :param workspace:   The workspace to convert.
+        :return:
+        """
+
+        x_unit = workspace.getAxis(0).getUnit().unitID()
+        y_unit = workspace.getAxis(1).getUnit().unitID()
+
+        # ----- Quick Conversions (Wavelength and DeltaE) -----
+
+        if x_unit == 'Wavelength':
+            return self._clone_ws(workspace)
+        elif y_unit == 'Wavelength':
+            self._transposed = True
+            return self._tranpose_ws(workspace)
+        elif x_unit == 'EnergyTransfer':
+            return self._convert_units(workspace, "Wavelength", self._emode, self._efixed)
+        elif y_unit == 'EnergyTransfer':
+            self._transposed = True
+            workspace = self._tranpose_ws(workspace)
+            return self._convert_units(workspace, "Wavelength", self._emode, self._efixed)
+
+        # ----- Indirect Conversions -----
+
+        if self._emode == "Indirect":
+
+            # Check for Indirect Elastic (units of Q)
+            if y_unit == 'MomentumTransfer':
+                return self._create_waves_indirect_elastic(workspace)
+            elif x_unit == 'MomentumTransfer':
+                self._transposed = True
+                return self._create_waves_indirect_elastic(self._tranpose_ws(workspace))
+            return self._convert_units(workspace, "Wavelength", self._emode, self._efixed)
```

Comment:
It's this line that I wanted also to be `return self._create_waves_indirect_elastic(workspace)`. I.e. if it is indirect, and none of the units is `MomentumTransfer`, fall back to indirect elastic.
The rest seems fine.

Reply:
If neither axis are in Q, is this case not an Indirect Inelastic case?
In this situation, does it still make sense to use _create_waves_indirect_elastic?

Reply:
The IN16b elastic case can be x=Temperature & y=Spectrum

Reply:
How can elastic cases be distinguished from inelastic, or do they not need to be?
Indirect Inelastic cases will be picked up by create_waves_indirect_elastic by making this change.

Reply:
Inelastic will have x=EnergyTransfer

Reply:
If x (or y in case of transposition) is EnergyTransfer, workspace is currently converted to Wavelength using the ConvertUnits algorithm with EFixed, so it will never reach this point in the code. 
Are we considering any case where x (or y in case of transposition) isn't EnergyTransfer and emode is Indirect, to be Indirect Elastic?

Reply:
If emode is Indirect and x (or y) is not EnergyTransfer it is Indirect Elastic. ISIS then has Q and ILL Spectrum.